app/Services/ModuleService.php

```diff
@@ -70,4 +83,203 @@ public function getAdminLinks(): array
     {
         return self::$adminLinks;
     }
+
+    /**
+     * Get All modules from Database
+     *
+     * @return object
+     */
+    public function getAllModules(): object
+    {
+        return Module::all();
+    }
+
+    /**
+     * Get Module Information from Database.
+     *
+     * @param $id
+     *
+     * @return object
+     */
+    public function getModule($id): object
+    {
+        return Module::find($id);
+    }
+
+    /**
+     * Adding installed module to the database
+     *
+     * @param $module_name
+     *
+     * @return bool
+     */
+    public function addModule($module_name): bool
+    {
+        /*Check if module already exists*/
+        if (Module::where('name', $module_name)->exists()) {
+            throw new ModuleExistsException($module_name);
```

Comment:
Is this trapped on the by the caller and then `flash()` called there with the error from `$exception->getMessage()`?

Reply:
Oh, I didn't get it. 
I should call it by flash() and not Exception ?

Reply:
No, still call the Exception, but put a try/catch around where it's being called from